src/main/java/com/hotels/beeju/extensions/BeejuJUnitExtension.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.thrift.TException;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.beeju.core.BeejuCore;
+
+abstract public class BeejuJUnitExtension implements BeforeEachCallback, AfterEachCallback {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BeejuJUnitExtension.class);
+  public BeejuCore core;
+  private Path tempDirectory;
```

Comment:
can we use the new junit5 `@TempDir ` in the extension? (or does that only work in tests?)

Reply:
I remember having some issues when I first tried using the new @TempDir but I'm going to test some options again to see what works best

Reply:
https://github.com/junit-team/junit5/issues/1786
Yeah it looks like @TempDir isn't supported as a part of a custom extension, only on test classes. But a solution someone suggested was to use `Files.createDirectory` in the @BeforeEach in the extension and delete it in @AfterEach which seems like a neat solution.